### PR TITLE
v2 parser: change assignment's lhs to expression

### PR DESCRIPTION
**Pull request by saifalkatout**
### What was wrong?
The new parser (v2 branch) wrongly supposes the assignment destination is a pattern, but it should be an expression.

### How was it fixed?
Parse lhs in assignment statements as an expression.

### To-Do
- Write tests for PR

### CODE REVIEW

File: crates/hir/src/hir_def/stmt.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@ pub enum Stmt {
     Let(PatId, Option<TypeId>, Option<ExprId>),
     /// The `Assign` statement. The first `PatId` is the pattern for binding,
```
Comment (Y-Nak) [inline crates/hir/src/hir_def/stmt.rs:13]:
```suggestion
    /// The `Assign` statement. The first `ExprId` is the destination of the assignment,
```

**Review by Y-Nak — COMMENTED** [review]
LGTM.  I left one nit.

Also, would you remove [e898016](https://github.com/ethereum/fe/pull/935/commits/e898016d578b7adb83bc2ca2222705b1505e8c4a) merge commit?

**Comment by saifalkatout** [comment]
@Y-Nak how pivotal is the removal of unnecessary commits ? I tried to revert and rebase but nothing seemed to work, I can delete the branch and push the changes again if necessary.

**Comment by Y-Nak** [comment]
You can squash these commits(you probably need to use `git rabase -i` command). (NOTE: I can squash your changes on github, but I think this would be a nice opportunity for you to get used to git.)

**Comment by Y-Nak** [comment]
@saifalkatout oh, you didn't need to close this. You could squash, then force push.

**Comment by saifalkatout** [comment]
I am sort of experimenting with it, To see how I could revert back. If you check my fork Im at 17 commits ahead 😂

**Pull request closed without merging**